content/learning-paths/laptops-and-desktops/mca-godbolt/running_mca.md

```diff
@@ -0,0 +1,350 @@
+
+Use a text editor to save the program below in a file named `sum_test2.s`:
+```
+add x10, x1, x2
+add x11, x3, x4
+add x12, x5, x6
+add x13, x10, x11
+add x14, x12, x13
+```
+
+Now run llvm-mca on this program using:
+`llvm-mca -mtriple=aarch64  -mcpu=neoverse-v2 sum_test2.s`
+
+The produced output looks like below:
+```
+Iterations:        100
+Instructions:      500
+Total Cycles:      88
+Total uOps:        500
+
+Dispatch Width:    16
+uOps Per Cycle:    5.68
+IPC:               5.68
+Block RThroughput: 0.8
+
+
+Instruction Info:
+[1]: #uOps
+[2]: Latency
+[3]: RThroughput
+[4]: MayLoad
+[5]: MayStore
+[6]: HasSideEffects (U)
+
+[1]    [2]    [3]    [4]    [5]    [6]    Instructions:
+ 1      1     0.17                        add   x10, x1, x2
+ 1      1     0.17                        add   x11, x3, x4
+ 1      1     0.17                        add   x12, x5, x6
+ 1      1     0.17                        add   x13, x10, x11
+ 1      1     0.17                        add   x14, x12, x13
+
+
+Resources:
+[0.0] - V2UnitB
+[0.1] - V2UnitB
+[1.0] - V2UnitD
+[1.1] - V2UnitD
+[2]   - V2UnitL2
+[3.0] - V2UnitL01
+[3.1] - V2UnitL01
+[4]   - V2UnitM0
+[5]   - V2UnitM1
+[6]   - V2UnitS0
+[7]   - V2UnitS1
+[8]   - V2UnitS2
+[9]   - V2UnitS3
+[10]  - V2UnitV0
+[11]  - V2UnitV1
+[12]  - V2UnitV2
+[13]  - V2UnitV3
+
+
+Resource pressure per iteration:
+[0.0]  [0.1]  [1.0]  [1.1]  [2]    [3.0]  [3.1]  [4]    [5]    [6]    [7]    [8]    [9]    [10]   [11]   [12]   [13]
+ -      -      -      -      -      -      -     0.83   0.83   0.83   0.83   0.84   0.84    -      -      -      -
+
+Resource pressure by instruction:
+[0.0]  [0.1]  [1.0]  [1.1]  [2]    [3.0]  [3.1]  [4]    [5]    [6]    [7]    [8]    [9]    [10]   [11]   [12]   [13]   Instructions:
+ -      -      -      -      -      -      -     0.17   0.48   0.17   0.17    -     0.01    -      -      -      -     add      x10, x1, x2
+ -      -      -      -      -      -      -     0.48   0.17   0.18   0.16   0.01    -      -      -      -      -     add      x11, x3, x4
+ -      -      -      -      -      -      -     0.17   0.18   0.16   0.33   0.16    -      -      -      -      -     add      x12, x5, x6
+ -      -      -      -      -      -      -      -      -     0.32   0.17   0.49   0.02    -      -      -      -     add      x13, x10, x11
+ -      -      -      -      -      -      -     0.01    -      -      -     0.18   0.81    -      -      -      -     add      x14, x12, x13
+```
+
+We can immediately see an improvement in the performance of our program by looking a the number of total cycles it took to execute and the IPC. Below you can see a comparison between the MCA performance metrics of `sum_test1.s` and `sum_test2.s`:
+
+|                   | sum_test1 | sum_test2 |
+| ----------------- | --------- | --------- |
+| Iterations        | 100       | 100       |
+| Instructions      | 500       | 300       |
```

Comment:
Is it really 300 in the sum_test2 column ? My own experiment shows 500 --- which makes sense to me.

Reply:
You are right, this should be 500. I changed the code and forgot to update the table with the instruction number. My bad.

Reply:
Should be correct now.

Reply:
Thanks